src/manifolds/MultinomialDoublyStochastic.jl

```diff
@@ -0,0 +1,286 @@
+"""
+check_manifold_point(::MultinomialDoubleStochasticMatrices, ::Any)
+function check_manifold_point(
+    M::MultinomialDoubleStochasticMatrices{n},
+    p;
+    kwargs...,
+) where {n}
+    mpv =
+        invoke(check_manifold_point, Tuple{supertype(typeof(M)),typeof(p)}, M, p; kwargs...)
+    mpv === nothing || return mpv
+    c = sum(p, dims = 1) # due to symmetry we only have to check the cols
+    if !isapprox(norm(c - ones(1, n)), 0.0; kwargs...)
+        return DomainError(
+            c,
+            "The point $(p) does not lie on $M, since its columns do not sum up to one.",
+        )
+    end
+    r = sum(p, dims = 2) # due to symmetry we only have to check the cols
+    if !isapprox(norm(r - ones(n, 1)), 0.0; kwargs...)
+        return DomainError(
+            r,
+            "The point $(p) does not lie on $M, since its rows do not sum up to one.",
+        )
+    end
+    if !(minimum(p) > 0)
+        return DomainError(
+            minimum(p),
+            "The point $(p) does not lie on $M, since at least one of its entries is nonpositive.",
+        )
+    end
+    return nothing
+end
+@doc raw"""
+    check_tangent_vector(M::MultinomialDoubleStochasticMatrices p, X; check_base_point = true, kwargs...)
+
+Checks whether `X` is a valid tangent vector to `p` on the [`MultinomialDoubleStochasticMatrices`](@ref) `M`.
+This means, that `p` is valid, that `X` is of correct dimension and sums to zero along any
+column or row.
+
+The optional parameter `check_base_point` indicates, whether to call
+[`check_manifold_point`](@ref check_manifold_point(::MultinomialDoubleStochasticMatrices, ::Any))  for `p`.
+"""
+function check_tangent_vector(
+    M::MultinomialDoubleStochasticMatrices{n},
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+) where {n}
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{supertype(typeof(M)),typeof(p),typeof(X)},
+        M,
+        p,
+        X;
+        check_base_point = false, # already checked above
+        kwargs...,
+    )
+    mpv === nothing || return mpv
+    c = sum(X, dims = 1) # due to symmetry, we only have to check columns
+    if !isapprox(norm(c), 0.0; kwargs...)
+        return DomainError(
+            c,
+            "The matrix $(X) is not a tangent vector to $(p) on $(M), since its columns do not sum up to zero.",
+        )
+    end
+    r = sum(X, dims = 2) # due to symmetry, we only have to check columns
+    if !isapprox(norm(r), 0.0; kwargs...)
+        return DomainError(
+            r,
+            "The matrix $(X) is not a tangent vector to $(p) on $(M), since its rows do not sum up to zero.",
+        )
+    end
+    return nothing
+end
+
+function decorated_manifold(::MultinomialDoubleStochasticMatrices{N}) where {N}
+    return Euclidean(N, N; field = ℝ)
```

Comment:
It could also be `MultinomialMatrices`. Do we have any general guidelines about handling such hierarchies of embeddings? Do we want to put something more specific here first?

Reply:
Indeed, I have to check; probably one has to have one check less (Multinomial checks for stochastic columns), one can reuse inner, sounds good.

In general I would give the rule of thumb: Use an embedding that gives you the most possible reuse of functions.

Reply:
I noticed that it changed a few things
* while column unit sum is not checked here, we only check rows
* similarly for the symmetric case, switching to `MultinomialMatrices` avoids positivity check and columns check (and hence row by symmetry) which are all done in the embedding, but symmetric is checked now therein (using the check from `SymmetricMatrices(n)` explicitly to also reuse code there).

Reply:
I further notices that with an abstract type (by symmetry both are doubly stochastic) one could even avoid reimplementing projection of a point, so thanks for that hint.